libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,274 @@
+
+        setInPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('inPointSelected');
+    };
+
+    const markOutPoint = () => {
+        setErrorMessage('');
+
+        console.log({ outpoint: audioRef.current!.currentTime });
+
+        const timeRangeSelection = {
+            inPointSeconds: inPointSeconds as number,
+            outPointSeconds: audioRef.current!.currentTime as number,
+        };
+
+        const duration = audioRef.current!.duration;
+
+        if (!isValidTimeRangeSelection(timeRangeSelection, duration)) {
+            if (timeRangeSelection.inPointSeconds >= duration) {
+                setInPointSeconds(null);
+
+                setOutPointSeconds(null);
+
+                setUxState({ timeRangeSelectionBar: null });
+
+                onTimeRangeSelected(null);
+
+                setErrorMessage(
+                    'Not a valid time range.  Marker points cannot be set after play ends'
+                );
+            } else {
+                setErrorMessage('Not a valid time range.  Out point must come after in point');
+            }
+
+            return;
+        }
+
+        setOutPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('fullSelection');
+
+        onTimeRangeSelected(timeRangeSelection);
+    };
+
+    const clearMarkers = () => {
+        setInPointSeconds(null);
+
+        setOutPointSeconds(null);
+
+        setUxState({ timeRangeSelectionBar: null });
+
+        onTimeRangeSelected(null);
+
+        setErrorMessage('');
+    };
+
+    useKeyDown(() => {
+        markInPoint();
+    }, [KeyboardKey.inpoint]);
+
+    useKeyDown(() => {
+        markOutPoint();
+    }, [KeyboardKey.outpoint]);
+
+    useKeyDown(() => {
+        clearMarkers();
+    }, [KeyboardKey.clear]);
+
+    const toggleTimeRangeSelectionBars = (timeRangeSelectionBar: TimeRangeSelectionBar) => {
+        console.log(`timeRangeSelectionBar: ${timeRangeSelectionBar} uxstate: `, uxState);
+
+        const isVisible = isBarVisible(timeRangeSelectionBar, uxState);
+
+        if (isVisible) {
+            setUxState({ timeRangeSelectionBar: null });
+        } else {
+            setUxState({ timeRangeSelectionBar: timeRangeSelectionBar });
+        }
+    };
+
+    return (
+        <Stack>
+            <StyledAudioPlayer ref={audioRef} onTimeUpdate={onTimeUpdate} controls>
+                {isAudioMIMEType(mimeType) ? (
+                    <source key={mimeType} src={audioUrl} type={mimeType} />
+                ) : (
+                    <>
+                        {Object.values(AudioMIMEType).map((mimeType) => (
+                            <source key={mimeType} src={audioUrl} type={mimeType} />
+                        ))}
+                    </>
+                )}
+                Your browser does not support the audio element.
+            </StyledAudioPlayer>
+            <Box sx={{ height: '12px' }}>
+                {errorMessage !== '' ? (
```

Comment:
This design could be improved eventually.  Not sure if we'll have a central place for error messages or if it should be tied to the player